runners/apex/src/main/java/org/apache/beam/runners/apex/ApexRunner.java

```diff
@@ -139,12 +140,8 @@ public void populateDAG(DAG dag, Configuration conf) {
       lc.runAsync();
       if (options.getRunMillis() > 0) {
         try {
-          long timeout = System.currentTimeMillis() + options.getRunMillis();
-          while (System.currentTimeMillis() < timeout) {
-            if (assertionError != null) {
-              throw assertionError;
-            }
-          }
+          // this is necessary for tests that just call run() and not waitUntilFinish
+          ApexRunnerResult.waitUntilFinished(lc, Duration.millis(options.getRunMillis()));
```

Comment:
runMillis is only set in the test runner. And yes, I think this can be now be moved over to the test runner, should I do it as part of this PR?